web/src/ui/nodes/properties/author/index.ts

```diff
@@ -0,0 +1,183 @@
+ * typing and because they are all simple atomic values.
+ */
+@customElement('stencila-ui-node-author')
+@withTwind()
+export class UINodeAuthor extends LitElement {
+  /**
+   * The type of node that is the author
+   *
+   * Might be useful for determining default icons etc.
+   */
+  @property()
+  type: 'Person' | 'Organization' | 'SoftwareApplication'
+
+  /**
+   * The id of the author
+   *
+   * Currently only available for `SoftwareApplication` authors. The intension is
+   * to use this to be able to fetch more information about the authors (e.g. affiliation)
+   * from some canonical source in the document. But currently this has not been finalized
+   * (or even started).
+   */
+  @property({ attribute: '_id' })
+  $id?: string
+
+  /**
+   * The name of the author
+   *
+   * Should be available for all author types, for a `Person` by concatenating
+   *  `givenNames` and `familyNames`.
+   */
+  @property()
+  name: string
+
+  /**
+   * The role that the author has
+   *
+   * Only available for `AuthorRole` authors.
+   */
+  @property({ attribute: 'role' })
+  roleName?: AuthorRoleName
+
+  /**
+   * The timestamp of the last modification made by the author in a particular role
+   *
+   * Only available for `AuthorRole` authors.
+   */
+  @property({ type: Number })
+  timestamp?: number
+
+  /**
+   * Additional details to display - e.g. a Person's organisation.
+   */
+  @property()
+  details?: string
+
+  override render() {
+    return html`<div class="flex flex-row gap-x-2">
+      <div class="flex items-center justify-center">
+        <div class="w-6 h-6 flex items-center justify-center grow-0 stretch-0">
+          ${this.getIconOrAvatar()}
+        </div>
+      </div>
+      <div class="grow flex flex-col justify-center">
+        <span class=${`text-2xs leading-none ${this.roleName ? '' : 'hidden'}`}
+          >${this.roleName}</span
+        >
+        <span
+          class="text-xs leading-5 overflow-hidden whitespace-nowrap text-ellipsis inline-block"
+          >${this.name}</span
+        >
+        <span
+          class=${`text-2xs leading-none overflow-hidden whitespace-nowrap text-ellipsis inline-block ${this.details ? '' : 'hidden'}`}
+          >${this.details}</span
+        >
+      </div>
+      <div
+        class=${`grow-0 shrink-0 text-2xs text-right ${this.roleName ? 'pt-3' : 'pt-1'}`}
+      >
+        <stencila-ui-node-timestamp-property
+          timestamp=${this.timestamp}
+        ></stencila-ui-node-timestamp-property>
+      </div>
+    </div>`
+  }
+
+  /**
+   * Take the object's ID & split into application name & version. This assumes
+   * that a software application's $id has a format of `name/version` - where
+   * we have a forward slash to split strings from.
+   */
+  private splitID() {
+    return this.$id?.trim().split('/') ?? []
+  }
+
+  private getInitial() {
+    return this.name.charAt(0)
+  }
+
+  private getSoftwareIcon() {
```

Comment:
A bit picky, but I wonder if all these functions that `return html` should be prefixed with `render` instrad of `get`

Reply:
Done